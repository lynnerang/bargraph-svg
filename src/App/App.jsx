import React, { useState } from 'react';
import Slider from '@material-ui/core/Slider';
import Bargraph from '../Bargraph/Bargraph';
import Dial from '../Dial/Dial';

const App = () => {
  const [mv, setMv] = useState(0);
  const [inputVal, setInputVal] = useState(0);

  const onUpdate = (val) => {
    setMv(val);
    setInputVal(val);
  }

  const onSubmit = (e) => {
    e.preventDefault();
    onUpdate(inputVal);
  }

  const onDialChange = (val) => {
    // console.log("val " + val)
    const diff = Math.abs(mv - val);
    // console.log("diff " + diff)
    let newVal = mv + diff;
    // console.log("newVal " + newVal)
    if (newVal < 5000) newVal = 5000;
    else if (newVal < 0) newVal = 0;

    setMv(newVal);
  }

  const styles = {
    fontFamily: "sans-serif",
    textAlign: "center",
    marginTop: "6rem"
  };

  console.log(mv)

  return (
    <div className="App">
      <header className="App-header">
        <h1>Bargraph Simulator 5000</h1>
        <div style={styles}>
          <Dial
            width={200}
            height={200}
            onDialChange={onDialChange}
            value={mv}
          />
        </div>
      </header>
      <main>
        <Slider
          orientation="vertical"
          defaultValue={0}
          value={parseInt(mv)}
          onChange={(e, val) => onUpdate(val)}
          aria-labelledby="slider"
          min={0}
          max={5000}
        />
        <form className="mv-entry" onSubmit={onSubmit}>
          <label htmlFor="mv-input">mV (millivolts)</label>
          <div>
            <input type="number" id="mv-input" value={inputVal} onChange={(e) => setInputVal(e.target.value)}/>
            <input type="submit" className="submit-btn" value="Go"/>
          </div>
        </form>
        <Bargraph power={(mv / 50) / 20}/>
      </main>
    </div>
  );
}

export default App;
