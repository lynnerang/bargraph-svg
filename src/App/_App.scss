* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.App {
  text-align: center;
}

main {
  height: 200px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 40px;
}

.mv-entry {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  
  #mv-input {
    -webkit-appearance: none;
    width: 80px;
    height: 35px;
    border-radius: 5px 0 0 5px;
    border: 1px inset silver;
    padding: 8px;
    font-size: 16px;
  }
  
  .submit-btn {
    -webkit-appearance: none;
    width: 40px;
    height: 35px;
    font-size: 16px;
    border-radius: 0 5px 5px 0;
    border: none;
    background: rgb(125, 125, 191);
    color: white;
    font-weight: 500;
    cursor: pointer;
  
    &:hover {
      background: rgb(155, 155, 235);
    }
  }
  
  label {
    font-size: 14px;
    margin-bottom: 10px;
  }
}
