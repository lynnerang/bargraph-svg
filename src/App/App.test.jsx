import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { createMount } from '@material-ui/core/test-utils';
import Slider from "@material-ui/core/Slider";

describe('App', () => {
  let mount;

  beforeEach(() => {
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });


  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const wrapper = mount(<App />);
    expect(wrapper).toMatchSnapshot();
  })

  it('displays an initial volt count of zero', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find('.volts-screen').text()).toEqual('0.000');
  })

  it('updates the volts displayed when the slider value changes', () => {
    const wrapper = mount(<App />);
    wrapper.find(Slider).at(0).simulate('change', { target: { value: 500 } });
    setTimeout(() => expect(wrapper.find('.volts-screen').text()).toEqual('0.500'), 0);
  })

});