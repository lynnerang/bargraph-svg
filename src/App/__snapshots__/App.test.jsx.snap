// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <h1>
        Bargraph Simulator 5000
      </h1>
    </header>
    <main>
      <section
        className="controls"
      >
        <div
          className="controls-inner"
        >
          <article
            className="dials-wrapper"
          >
            <Dial
              name="Coarse"
              onDialChange={[Function]}
            >
              <article
                className="dial-wrapper"
              >
                <button>
                  <i
                    className="fas fa-caret-up"
                    onClick={[Function]}
                  />
                </button>
                <div
                  className="dial-bg"
                >
                  <h2
                    className="dial-name"
                  >
                    Coarse
                  </h2>
                  <p
                    className="dial-label"
                  >
                    (
                    100
                     mV)
                  </p>
                </div>
                <button>
                  <i
                    className="fas fa-caret-down"
                    onClick={[Function]}
                  />
                </button>
              </article>
            </Dial>
            <Dial
              name="Fine"
              onDialChange={[Function]}
            >
              <article
                className="dial-wrapper"
              >
                <button>
                  <i
                    className="fas fa-caret-up"
                    onClick={[Function]}
                  />
                </button>
                <div
                  className="dial-bg"
                >
                  <h2
                    className="dial-name"
                  >
                    Fine
                  </h2>
                  <p
                    className="dial-label"
                  >
                    (
                    1
                     mV)
                  </p>
                </div>
                <button>
                  <i
                    className="fas fa-caret-down"
                    onClick={[Function]}
                  />
                </button>
              </article>
            </Dial>
          </article>
          <article
            className="slider-wrapper"
          >
            <p
              className="slider-text"
            >
              <span
                className="slider-label"
              >
                MAX
              </span>
              (5000mV)
            </p>
            <div
              className="slider-container"
            >
              <WithStyles(ForwardRef(Slider))
                aria-labelledby="slider"
                defaultValue={0}
                max={5000}
                min={0}
                name="slider-input"
                onChange={[Function]}
                orientation="vertical"
                value={0}
              >
                <ForwardRef(Slider)
                  aria-labelledby="slider"
                  classes={
                    Object {
                      "active": "MuiSlider-active",
                      "colorPrimary": "MuiSlider-colorPrimary",
                      "colorSecondary": "MuiSlider-colorSecondary",
                      "disabled": "Mui-disabled",
                      "focusVisible": "Mui-focusVisible",
                      "mark": "MuiSlider-mark",
                      "markActive": "MuiSlider-markActive",
                      "markLabel": "MuiSlider-markLabel",
                      "markLabelActive": "MuiSlider-markLabelActive",
                      "marked": "MuiSlider-marked",
                      "rail": "MuiSlider-rail",
                      "root": "MuiSlider-root",
                      "thumb": "MuiSlider-thumb",
                      "thumbColorPrimary": "MuiSlider-thumbColorPrimary",
                      "thumbColorSecondary": "MuiSlider-thumbColorSecondary",
                      "track": "MuiSlider-track",
                      "valueLabel": "MuiSlider-valueLabel",
                      "vertical": "MuiSlider-vertical",
                    }
                  }
                  defaultValue={0}
                  max={5000}
                  min={0}
                  name="slider-input"
                  onChange={[Function]}
                  orientation="vertical"
                  value={0}
                >
                  <span
                    className="MuiSlider-root MuiSlider-colorPrimary MuiSlider-vertical"
                    onMouseDown={[Function]}
                  >
                    <span
                      className="MuiSlider-rail"
                    />
                    <span
                      className="MuiSlider-track"
                      style={
                        Object {
                          "bottom": "0%",
                          "height": "0%",
                        }
                      }
                    />
                    <input
                      name="slider-input"
                      type="hidden"
                      value="0"
                    />
                    <WithStyles(ValueLabel)
                      className="MuiSlider-valueLabel"
                      disabled={false}
                      index={0}
                      key="0"
                      open={false}
                      value={0}
                      valueLabelDisplay="off"
                      valueLabelFormat={[Function]}
                    >
                      <ValueLabel
                        className="MuiSlider-valueLabel"
                        classes={
                          Object {
                            "circle": "PrivateValueLabel-circle-45",
                            "label": "PrivateValueLabel-label-46",
                            "offset": "PrivateValueLabel-offset-44",
                            "open": "PrivateValueLabel-open-43",
                            "thumb": "PrivateValueLabel-thumb-42",
                          }
                        }
                        disabled={false}
                        index={0}
                        open={false}
                        value={0}
                        valueLabelDisplay="off"
                        valueLabelFormat={[Function]}
                      >
                        <span
                          aria-labelledby="slider"
                          aria-orientation="vertical"
                          aria-valuemax={5000}
                          aria-valuemin={0}
                          aria-valuenow={0}
                          className="MuiSlider-thumb MuiSlider-thumbColorPrimary"
                          data-index={0}
                          onBlur={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          role="slider"
                          style={
                            Object {
                              "bottom": "0%",
                            }
                          }
                          tabIndex={0}
                        />
                      </ValueLabel>
                    </WithStyles(ValueLabel)>
                  </span>
                </ForwardRef(Slider)>
              </WithStyles(ForwardRef(Slider))>
            </div>
            <p
              className="slider-text"
            >
              <span
                className="slider-label"
              >
                MIN
              </span>
              (0mV)
            </p>
          </article>
          <article
            className="volts-wrapper"
          >
            <p
              aria-live="polite"
              className="volts-screen"
            >
              0.000
            </p>
            <p
              className="volts-label"
            >
               V (Volts)
            </p>
          </article>
        </div>
      </section>
      <section
        className="wires"
      />
      <section
        aria-hidden={true}
        className="bargraph"
      >
        <Bargraph
          power={0}
        >
          <SvgLoader
            className="power-0"
            onSVGReady={[Function]}
            path="o9-Bargraph-01.svg"
            style={null}
            svgXML={null}
          >
            <ReactSVG
              callback={[Function]}
              className="power-0"
              path="o9-Bargraph-01.svg"
              style={null}
              svgXML={null}
            >
              <svg
                className="power-0"
                data-src="o9-Bargraph-01.svg"
                style={null}
              />
            </ReactSVG>
          </SvgLoader>
        </Bargraph>
      </section>
    </main>
  </div>
</App>
`;
